File: backend/app/schemas/hotspots/hotspots.py
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class HotspotResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    ebird_id: str
    name: str
    source: str
    latitude: float
    longitude: float


class HotspotListResponse(BaseModel):
    items: list[HotspotResponse]
    total: int
    page: int
    page_size: int

File: frontend/src/types/hotspots.ts
export interface Hotspot {
  id: string;
  ebird_id: string;
  name: string;
  source: string;

  latitude: number;
  longitude: number;
}

export interface NearbyHotspot extends Hotspot {
  distance_km: number;
}

export interface HotspotListResponse {
  items: Hotspot[];
  total: number;
  page: number;
  page_size: number;
}

export interface NearbyHotspotListResponse {
  items: NearbyHotspot[];
  total: number;
  page: number;
  page_size: number;
}